src/translator/history.h

```diff
@@ -37,7 +37,7 @@ class History {
 
   size_t size() const { return history_.size(); } // number of time steps
 
-  NBestList nBest(size_t n) const {
+  NBestList nBest(size_t n, bool skip_empty = false) const {
```

Comment:
Can `skip_empty = true` produce an n-best list with fewer than n elements and there is no longer a guarantee that all n-best lists are of the same size? If so, this requires a comment or even a warning. Could you add a docstring to this method?

Reply:
Yes, if n equals the beam size. I've added a docstring with a note to that effect. I think a warning would be overkill, as skipEmpty is not the default behaviour. The typical use case for this new option is calling nBest(1, true)[0] with a beam size > 1.

Reply:
Yes, sorry, I meant a warning in the comment, your note serves the purpose.